### PR TITLE
MAINT: clearer error messages for refguide check.

### PR DESCRIPTION
**Pull request by rgommers**
As suggested here https://github.com/scipy/scipy/pull/6763#issuecomment-258679685

### CODE REVIEW

File: tools/refguide_check.py [hunk 1]
```diff
@@ -272,11 +272,17 @@ def check_items(all_dict, names, deprecated, others, module_name, dots=True):
...
+            output += "\nThis issue can be fix by adding these objects to the\n"
+            output += "function listing in __init__.py for this module\n"
+
         if len(only_ref) > 0:
             output += "ERROR: objects in refguide but not in %s.__all__::\n\n" % module_name
             for name in sorted(only_ref):
                 output += "    " + name + "\n"
 
+            output += "\nThis issue should likely be fixed by removing these objects\n"
+            output += "from the function listing in __init__.py for this module\n"
```
Comment (ev-br) [inline tools/refguide_check.py]: or adding them to `__all__`.

File: tools/refguide_check.py [hunk 2]
```diff
@@ -272,11 +272,17 @@ def check_items(all_dict, names, deprecated, others, module_name, dots=True):
             for name in sorted(only_all):
                 output += "    " + name + "\n"
 
+            output += "\nThis issue can be fix by adding these objects to the\n"
```
Comment (ev-br) [inline tools/refguide_check.py]: typo fix*ed*

**Comment by Eric89GXL** [comment]
LGTM after addressing @ev-br's comments

**Comment by rgommers** [comment]
all fixed

**Comment by ev-br** [comment]
Merging, thanks Ralf

**Pull request merged**